### DISCUSSION TITLE
Prevent deletion of files in media gallery, if they are used somewhere

### DISCUSSION THREAD

**Discussion by LasseRosenow** [directus/directus#10967] [Feature Requests]
Currently it is possible to delete files from the media gallery even though they are used somewhere.

I think this situation is not ideal, as I would like to prevent users from accidentally delete files that they think they don't need and end up causing issues somewhere unexpected.

There was a issue, but it was closed, as it was considered not a BUG, but a FEATURE REQUEST: #10923 
So feel free to read through the issue as we had a good discussion there :)

----

I would like to quote @Oreilles last post as it states the result of our discussion:

> We won't be able to find whether an image is reference somewhere before trying to delete it, because that would be a massive overhead (you'd have to iterate over every row of collection to find out whether a file field references the corresponding image).

> I agree however that being able to pick the correct on_delete behaviour (prevent deletion, cascade, or set null) for file/image fields, just like it is the case for other m2o fields, might be a useful, but since it is not a bug in itself I'll leave it up to you to open a new Discussion if that is something you would want. 👍

> A better error message would be nice too - but since there doesn't appear to be an actual bug here (the database only prevented a forbidden null value), I will close this issue.

**Comment by rijkvanzanten** [comment]
https://github.com/directus/directus/pull/10985 adds in the missing Relations tab to the advanced field flow for file type fields to allow you configure the relational trigger 👍🏻 

<img width="1033" alt="CleanShot 2022-01-11 at 17 01 01@2x" src="https://user-images.githubusercontent.com/9141017/149028287-c20bf324-b68b-4c96-bbc0-2aca30b56e7c.png">
